### PR TITLE
wayland: hack around erronous playback speed on overlapping monitors

### LINKED ISSUE #10489: Playback speed doubles when window is on two monitors
- mpv 0.34.0-395-g9add44b11a
- Arch Linux
- compiled from source (latest master)
- KDE Plasma 5.25.4 on Wayland
- Mesa 22.1.4 using AMD R9 380

I have two 60Hz monitors and when the window overlaps both monitors, the playback speed usually (but not always) gets doubled.

In this video it doesn't happen in the beginning, but then afterwards it does two times.
I only recorded the left monitor, as that's where where most of the window is, but moving it only a little bit onto the right one already triggers it.

https://user-images.githubusercontent.com/8932183/182983850-8b972924-158f-46cb-8c15-09d2d50e285d.mp4

### Reproduction steps
`mpv --no-config --video-sync=display-resample  https://github.com/haasn/interpolation-samples/raw/master/60.000.mkv`
Also happens with local files and files with different refresh rates.
Happens with all `video-sync=display-*` options.

### Expected behavior
Always plays video at the same speed, regardless of position
### Actual behavior
Playback at 2x speed when window is on two monitors
### Log file

[output.txt](https://github.com/mpv-player/mpv/files/9264531/output.txt)

### Sample files
[haasns 60 fps test video](https://github.com/haasn/interpolation-samples/raw/master/60.000.mkv)

### PR DESCRIPTION
**Pull request by Dudemanguy**
I haven't tested this yet, but this should fix #10489 assuming I did the math right.

### CODE REVIEW

File: video/out/wayland_common.c [hunk 1]
```diff
@@ -2096,6 +2099,19 @@ void vo_wayland_wait_frame(struct vo_wayland_state *wl)
     if (!wl->presentation && !wl_display_get_error(wl->display))
         wl_display_roundtrip(wl->display);
 
+    /* Insane hack for the edge case of mpv overlapping multiple monitors and
+     * while using a display-* mode. The poll wait may wake up early by at least 
+     * one frame on some compositors. */
```
Comment (sfan5) [inline video/out/wayland_common.c:2104]: can't it tell you which display it woke up for?

**Comment by Dudemanguy** [comment]
The idea behind this works but it actually waits too long. Also it actually only happens on KWin so I might consider this "not our bug" and just call it quits.

**Comment by christoph-heinrich** [comment]
Since this only happens on KWin and only when the window overlaps monitors, I agree that adding a workaround just for that is not really necessary (after all who actually watches a video over multiple monitors).
I think any serious compositor should support the presentation time protocol anyway.
Does anyone know of a tracking issue of a PR for adding this to Kwin, or is nobody even working on this? I've looked for for it a while ago but couldn't find anything.

**Comment by Dudemanguy** [comment]
Yeah don't think we want this.

**Pull request closed without merging**